src/engine/SpatialJoin.cpp

```diff
@@ -65,53 +53,35 @@ SpatialJoin::SpatialJoin(
 }
 
 // ____________________________________________________________________________
-void SpatialJoin::parseMaxDistance() {
-  const std::string& input = triple_.p_._iri;
-
-  if (ctre::match<MAX_DIST_IN_METERS_REGEX>(input)) {
-    std::string number = input.substr(
-        MAX_DIST_IN_METERS.size(),
-        input.size() - MAX_DIST_IN_METERS.size() - 1);  // -1: compensate for >
-    maxDist_ = std::stoll(number);
-  } else {
-    AD_THROW(
-        "parsing of the maximum distance for the "
-        "SpatialJoin operation was not possible");
-  }
-
-  if (maxDist_ < 0) {
-    AD_THROW("the maximum distance between two objects must be >= 0");
-  }
-}
+void SpatialJoin::parseConfigFromTriple() {
+  // Helper to convert a ctre match to an integer
+  auto matchToInt = [](std::string_view match) -> std::optional<long long> {
+    if (match.size() > 0) {
+      double res = 0;
+      absl::from_chars(match.data(), match.data() + match.size(), res);
```

Comment:
as you only support integers anyway, why not
`long long res; fromChars(..., res); return res`  whithout the static_cast?

Reply:
The double and cast is due to the (weird) behavior of `absl::from_chars`:

> Workalike compatibility version of std::from_chars from C++17. Currently this only supports the double and float types.